velox/expression/CastExpr.cpp

```diff
@@ -439,6 +439,10 @@ VectorPtr CastExpr::applyDecimal(
       applyIntToDecimalCastKernel<int32_t, toDecimalType>(
           rows, input, context, toType, castResult);
       break;
+    case TypeKind::DOUBLE:
+      applyDoubleToDecimal<double, toDecimalType>(
```

Comment:
We may also need to document this casting behavior in https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/presto/conversion.rst.